test/validators/custom-styles-validator.test.ts

```diff
@@ -0,0 +1,83 @@
+import * as path from 'path';
+import { CustomStylesValidator } from '../../lib/validators/custom-styles-validator';
+
+describe('CustomStylesValidator', () => {
+    let definition: any;
+    let error: jasmine.Spy;
+    let validator: CustomStylesValidator;
+
+    beforeEach(() => {
+        // valid definition
+        definition = {
+            customStyles: [],
+        };
+        error = jasmine.createSpy('error');
+    });
+
+    describe('validate', () => {
+        it('should do nothing when no custom styles are given', async () => {
+            delete definition.customStyles;
+            validator = new CustomStylesValidator(error, definition, new Set());
+            await validator.validate();
+            expect(error).not.toHaveBeenCalled();
+        });
+
+        it('should do nothing when no custom styles are given', async () => {
+            validator = new CustomStylesValidator(error, definition, new Set());
+            await validator.validate();
+            expect(error).not.toHaveBeenCalled();
+        });
+
+        it(`should throw no errors when the style doesn't have a default defined`, async () => {
+            definition.customStyles.push({
+                key: 'test-style',
+                label: 'Test Style',
+                type: 'JSON',
+            });
+            validator = new CustomStylesValidator(error, definition, new Set());
+            await validator.validate();
+            expect(error).not.toHaveBeenCalled();
+        });
+
+        it(`should throw now errors when the default file is available`, async () => {
+            definition.customStyles.push({
+                key: 'test-style',
+                label: 'Test Style',
+                type: 'JSON',
+                default: './styles/default-custom-styles/existing-file.json',
```

Comment:
Wouldn't it make more sense if this is placed in the `customStyles ` folder? Also naming is inconsistent. One has dashes, other one camelCase.

Reply:
The default styles aren't allowed in the `styles/customStyles` folder for syncing reasons. I now added them to a defaultCustomStyles folder in the root which should be fine. The reason for this logic was this comment: https://github.com/WoodWing/ecs-backend/pull/874#pullrequestreview-661823134